staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -404,7 +404,15 @@ type eventBroadcasterAdapterImpl struct {
 
 // NewEventBroadcasterAdapter creates a wrapper around new and legacy broadcasters to simplify
 // migration of individual components to the new Event API.
+//
+//logcheck:context // NewEventBroadcasterAdapterWithContext should be used instead because record.NewBroadcaster is called and works better when a context is supplied (contextual logging, cancelation).
 func NewEventBroadcasterAdapter(client clientset.Interface) EventBroadcasterAdapter {
+	return NewEventBroadcasterAdapterWithContext(context.Background(), client)
+}
+
+// NewEventBroadcasterAdapterWithContext creates a wrapper around new and legacy broadcasters to simplify
+// migration of individual components to the new Event API.
+func NewEventBroadcasterAdapterWithContext(ctx context.Context, client clientset.Interface) EventBroadcasterAdapter {
```

Comment:
This pattern is a bit confusing IMHO, passing a context to a constructor that has Start() and Stop() methods, I expect those methods to govern the lifecycle of the object created and not depend on the context used during initialization

Reply:
/cc @dgrisonnet

Reply:
This also appears in some of my other pending PRs, for example https://github.com/kubernetes/kubernetes/pull/122145/files

On the other hand, in several of those cases the object gets constructed by the same function which then also starts it, so the context is often (but not always!) readily available.

In https://github.com/kubernetes/kubernetes/pull/122148, some controllers accepted a context already. Some only took a logger and had to be converted to context.

Reply:
This makes sense to me given we are only exposing the same behavior that we inherit from the `record` package. This is the only discernible change between the `NewEventBroadcasterAdapter*` constructors which has been explicitly pointed out in the signature.